content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-curl-and-postman.md

```diff
@@ -0,0 +1,333 @@
+
+* [EU Central](https://eu-central1-api.compute.cloud.hpe.com)   
+* [AP Northeast](https://eu-central1-api.compute.cloud.hpe.com) 
+* [US West](https://us-west2-api.compute.cloud.hpe.com)
+
+The request uses the **GET** method for the **/compute-ops/v1beta2/servers** resource to obtain the list of the available servers as described in the API reference.
+
+![blog figure3](/img/greenlake-com-ops-api-curl25.png)
+
+The command uses a header parameter with keyword **"Authorization:Bearer {token}**"
+
+**?limit=2** at the end of the URL is a query parameter to limit the response to a maximum of two servers as documented in the request pane of the *servers* resource:
+
+![blog figure4](/img/greenlake-com-ops-api-curl19.png)
+
+ As a result of the command, the API response lists two compute servers that are on boarded and assigned to the corresponding application for the user's account.
+
+![blog figure4](/img/greenlake-com-ops-api-curl15.png)
+
+To see the API response code, add -I at the end of the command:
+
+```shell
+curl -X GET https://us-west2-api.compute.cloud.hpe.com/compute-ops/v1beta2/servers?limit=2 -H "Authorization:Bearer <your access_token_here>” –I
+```
+
+![blog figure5](/img/greenlake-com-ops-api-curl2.png)
+
+The response code displays **200 OK.** 
+
+This status response code indicates that the request has succeeded. The complete list of response codes for each resource in the API reference is shown below:
+
+![blog figure6](/img/greenlake-com-ops-api-curl26.png)
+
+The JSON response provided by the API is not formatted, which makes it difficult to read and understand the JSON content, but a tool like [jq](https://stedolan.github.io/jq/)  can prettify the content. Just add “| jq” at the end of the command to get a better visual display.
+
+![blog figure7](/img/greenlake-com-ops-api-curl4.png)
+
+The API supports several query parameters depending on the resource type, such as filter, limit (maximum number of records to return), offset (resource offset to start the response from) and sort (order in which to return the resources in the collection).
+
+Filters provide the ability to limit the resources that take part in the action of a REST call. When a REST call includes a filter, the GET or DELETE action is restricted to a response that meets the filter requirements. Filters are specified by using the query parameter 'filter'.
+
+Here is a simple example of filtering:
+
+**GET <URI>?filter=powerState eq 'Off'**
+
+This example shows a simple filter. The resources returned by the query are limited to results with the attribute **powerState** and the value **Off**.
+
+To use a filter on a nested property name, the '**/**' separator can be specified as follows:
+
+**GET /compute-ops/v1beta2/servers?filter=hardware/model eq 'ProLiant DL365 Gen10 Plus'\*\***\
```

Comment:
should all of the below be in a code block? Doesn't show that way in the preview, but I'm not sure.
![image](https://user-images.githubusercontent.com/77496261/205346504-7e1efaef-2a4f-454f-9137-1df83a32d69e.png)

Reply:
nope. ok this way